third_party/nvfuser/csrc/fusion.cpp

```diff
@@ -686,13 +692,14 @@ void Fusion::aliasOutputToInput(Val* output, Val* input) {
   if (input->getDataType().value() != output->getDataType().value()) {
     output = castOp(input->getDataType().value(), output);
   }
-  // TODO: output should be marked at the end of fusion definition #1488
-  addOutput(output);
 
   TORCH_INTERNAL_ASSERT(
       isAliasCompatible(input, output),
       "The input and output values are not alias-compatible.");
   io_alias_[output] = input;
+
+  // TODO: output should be marked at the end of fusion definition #1488
+  addOutput(output);
```

Comment:
@jjsjann123 I switched the order these lines so that `io_alias_[output]` can be checked inside `addOutput` to avoid infinite recursion. I think this is kosher but just letting you know.

Reply:
Appreciate the heads up. This should be fine~~